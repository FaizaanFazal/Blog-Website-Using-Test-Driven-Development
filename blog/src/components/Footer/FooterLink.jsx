import React from 'react';

const FooterLink = ({ linksData }) => (
  <div className="ftr-item">
    <h4 className="ftr-item-title text-lg">{ linksData.title }</h4>
    <ul className="ftr-nav-links">
      {
        linksData.links.map((link, index) => (
          <li className="ftr-nav-item" key={`link${index}`}>
            <a data-testid="otherLinks" href="/" className="ftr-nav-link text-base">{ link }</a>
          </li>
        ))
        }
    </ul>
  </div>
);

export default FooterLink;
