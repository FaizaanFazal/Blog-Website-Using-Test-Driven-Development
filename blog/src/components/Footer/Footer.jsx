import React from 'react';
import FooterLink from './FooterLink';
import { linksData } from '../../data/blog-posts';

export default function Footer() {
  return (
    <footer title="footer" data-testid="footer" className="footer bg-blue-dark">
      <div className="container">
        <div className="ftr-list">
          <div className="ftr-item">
            <a href="#" className="ftr-brand text-white">
              Blogs
              {' '}
              <span className="ftr-brand-dot bg-white" />
            </a>

            <p className="text text-base">
              Build a modern and creative website with crealand
            </p>
            <ul className="social-links text-white flex flex-wrap text-gray text-base">
              <li><a data-testid="social" href="/">Twitter</a></li>
              <li><a data-testid="social" href="/">Linkedin</a></li>
              <li><a data-testid="social" href="/">Instagram</a></li>
              <li><a data-testid="social" href="/">Google</a></li>
              <li><a data-testid="social" href="/">YouTube</a></li>
            </ul>
          </div>

          <FooterLink props={linksData} />

        </div>
        <div data-testid="copyright" className="ftr-text text-center text-gray text-base">Copyright &copy; 2023. Crafted with love.</div>
      </div>
    </footer>
  );
}
