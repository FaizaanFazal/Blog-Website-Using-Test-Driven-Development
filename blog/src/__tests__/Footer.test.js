import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Footer from '../components/Footer/Footer';

describe('Unit tests for Footer Component', () => {
  it('Snapshot test for rendering footer', () => {
    const { asFragment } = render(<MemoryRouter><Footer /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Dom test for-> renders the Footer', () => {
    render(<MemoryRouter><Footer /></MemoryRouter>);
    const navbarElement = screen.getByTestId('footer');
    expect(navbarElement).toBeInTheDocument();
  });

  it('Semantic tag name is footer', () => {
    render(<MemoryRouter><Footer /></MemoryRouter>);
    const footerElement = screen.getByTitle('footer');
    expect(footerElement).not.toBeNull();
  });

  it('Number of Socail links 5', () => {
    render(<MemoryRouter><Footer /></MemoryRouter>);
    const tagElements = screen.getAllByTestId('social');
    expect(tagElements.length).toBeGreaterThan(0);
  });

  it('Number of other links to be 18', () => {
    render(<MemoryRouter><Footer /></MemoryRouter>);
    const otherLinks = screen.getAllByTestId('otherLinks');
    expect(otherLinks.length).toBeGreaterThan(0);
  });

  it('renders copyright statement', () => {
    render(<MemoryRouter><Footer /></MemoryRouter>);
    const copyright = screen.getByTestId('copyright');
    expect(copyright.textContent.length).toBeGreaterThan(0);
  });
});
