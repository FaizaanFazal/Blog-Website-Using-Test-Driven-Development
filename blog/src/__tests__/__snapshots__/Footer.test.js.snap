// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Footer Component Snapshot test for rendering footer 1`] = `
<DocumentFragment>
  <footer
    class="footer bg-blue-dark"
    data-testid="footer"
    title="footer"
  >
    <div
      class="container"
    >
      <div
        class="ftr-list"
      >
        <div
          class="ftr-item"
        >
          <a
            class="ftr-brand text-white"
            href="#"
          >
            heya 
            <span
              class="ftr-brand-dot bg-white"
            />
          </a>
          <p
            class="text text-base"
          >
            Build a modern and creative website with crealand
          </p>
          <ul
            class="social-links text-white flex flex-wrap text-gray text-base"
          >
            <li>
              <a
                data-testid="social"
                href="#"
              >
                Twitter
              </a>
            </li>
            <li>
              <a
                data-testid="social"
                href="#"
              >
                Linkedin
              </a>
            </li>
            <li>
              <a
                data-testid="social"
                href="#"
              >
                Instagram
              </a>
            </li>
            <li>
              <a
                data-testid="social"
                href="#"
              >
                Google
              </a>
            </li>
            <li>
              <a
                data-testid="social"
                href="#"
              >
                YouTube
              </a>
            </li>
          </ul>
        </div>
        <div
          class="ftr-item"
        >
          <h4
            class="ftr-item-title text-lg"
          >
            Read Our
          </h4>
          <ul
            class="ftr-nav-links"
          >
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Journals
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Articles
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Biograpahy
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Old records
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Audio version
              </a>
            </li>
          </ul>
        </div>
        <div
          class="ftr-item"
        >
          <h4
            class="ftr-item-title text-lg"
          >
            Products
          </h4>
          <ul
            class="ftr-nav-links"
          >
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Reading collection
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Library
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Broadcast
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Podcast
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Book Series
              </a>
            </li>
          </ul>
        </div>
        <div
          class="ftr-item"
        >
          <h4
            class="ftr-item-title text-lg"
          >
            Quick Links
          </h4>
          <ul
            class="ftr-nav-links"
          >
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Blogs
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Featured
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Latest
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Illustrations
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Latest
              </a>
            </li>
          </ul>
        </div>
        <div
          class="ftr-item"
        >
          <h4
            class="ftr-item-title text-lg"
          >
            More
          </h4>
          <ul
            class="ftr-nav-links"
          >
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                License
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                About Us
              </a>
            </li>
            <li
              class="ftr-nav-item"
            >
              <a
                class="ftr-nav-link text-base"
                data-testid="otherLinks"
                href="#"
              >
                Our policy
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="ftr-text text-center text-gray text-base"
        data-testid="copyright"
      >
        Copyright Â© 2023. Crafted with love.
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
